consensus/ethash/ethash.go

```diff
@@ -564,7 +573,12 @@ func NewFullFaker() *Ethash {
 
 // NewShared creates a full sized ethash PoW shared between all requesters running
 // in the same process.
-func NewShared() *Ethash {
+func NewShared(progpowNumber *big.Int) *Ethash {
+	ethashMu.Lock()
+	if sharedEthash == nil {
+		sharedEthash = New(Config{"", 3, 0, "", 1, 0, ModeNormal, progpowNumber}, nil, false)
```

Comment:
For a clearer explanation, I would like to:

```go
func NewShared(progpowNumber *big.Int) *Ethash {
	ethashMu.Lock()
	if sharedEthash == nil {
		sharedEthash = New(
			Config{
				CacheDir: 		"",
				CachesInMem:  	3,
				CachesOnDisk: 	0,
				"",
				DatasetsInMem:  1,
				DatasetsOnDisk: 0,
				ModeNormal,
				progpowNumber,
			},
			nil, false,
		)
	}
	ethashMu.Unlock()
	return &Ethash{shared: sharedEthash}
}
```

My code maybe wrong... :sweat_smile: Could you fix like this?

Reply:
If we do it like that, and ever modify `Config` to add a field, that field will be left to default. As it is right now, it will cause a compilation error, which is probably better.